Node seems to be killed when running in TravisCI

I just switched [my project](https://github.com/dadoonet/spring-elasticsearch) to elasticsearch-maven-plugin (which is great BTW) and I found that my elasticsearch node is getting killed apparently.

The build is working super fine locally. But fails in TravisCI.

For example this build: https://travis-ci.org/dadoonet/spring-elasticsearch/builds/210279724

I suspect this is caused by https://github.com/travis-ci/travis-ci/issues/6534 but I would expect elasticsearch barfing more before dying actually.

May be there is a workaround or may be we should just document that it can happen an link to the Travis issue in the doc?

Some logs:

```
[2017-03-12T15:59:20,974][INFO ][o.e.n.Node               ] [Q3beqyO] starting ...
[2017-03-12T15:59:21,133][WARN ][i.n.u.i.MacAddressUtil   ] Failed to find a usable hardware address from the network interfaces; using random bytes: 26:93:fc:18:6b:6a:2c:ed
[2017-03-12T15:59:21,203][INFO ][o.e.t.TransportService   ] [Q3beqyO] publish_address {127.0.0.1:9300}, bound_addresses {[::1]:9300}, {127.0.0.1:9300}
[2017-03-12T15:59:21,211][WARN ][o.e.b.BootstrapChecks    ] [Q3beqyO] max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]
[DEBUG] Sending GET request to http://localhost:9200/
[DEBUG] Sending GET request to http://localhost:9200/
[DEBUG] Sending GET request to http://localhost:9200/
[2017-03-12T15:59:24,271][INFO ][o.e.c.s.ClusterService   ] [Q3beqyO] new_master {Q3beqyO}{Q3beqyOTRNmkkow3pXMK0g}{biMDuBQdQD2bxga9m2Cm6Q}{127.0.0.1}{127.0.0.1:9300}, reason: zen-disco-elected-as-master ([0] nodes joined)
[2017-03-12T15:59:24,334][INFO ][o.e.h.HttpServer         ] [Q3beqyO] publish_address {127.0.0.1:9200}, bound_addresses {[::1]:9200}, {127.0.0.1:9200}
[2017-03-12T15:59:24,335][INFO ][o.e.n.Node               ] [Q3beqyO] started
[2017-03-12T15:59:24,349][INFO ][o.e.g.GatewayService     ] [Q3beqyO] recovered [0] indices into cluster_state
[DEBUG] Sending GET request to http://localhost:9200/
[DEBUG] Response with status code 200 and content: {  "name" : "Q3beqyO",  "cluster_name" : "elasticsearch",  "cluster_uuid" : "zYXkHQ0nRmSeZiXf9o04OQ",  "version" : {    "number" : "5.2.2",    "build_hash" : "f9d9b74",    "build_date" : "2017-02-24T17:26:45.835Z",    "build_snapshot" : false,    "lucene_version" : "6.4.1"  },  "tagline" : "You Know, for Search"}
[DEBUG] Elasticsearch has started
```

Then some tests are running well:

```
15:59:30,846 DEBUG [f.p.e.t.i.IndexElasticsearchUpdater] Index [twitter] doesn't exist. Creating it.
15:59:30,847 TRACE [f.p.e.t.i.IndexElasticsearchUpdater] createIndex([twitter])
15:59:30,848 TRACE [f.p.e.t.i.IndexElasticsearchUpdater] Found settings for index [twitter]: [{
}]
[2017-03-12T15:59:31,063][INFO ][o.e.c.m.MetaDataCreateIndexService] [Q3beqyO] [twitter] creating index, cause [api], templates [], shards [5]/[1], mappings []
15:59:31,459 TRACE [f.p.e.t.i.IndexElasticsearchUpdater] /createIndex([twitter])
15:59:31,462 TRACE [f.p.e.t.SettingsReader] Reading file [es/twitter/tweet.json]...
15:59:31,537 DEBUG [f.p.e.t.t.TypeElasticsearchUpdater] Type [twitter]/[tweet] doesn't exist. Creating it.
15:59:31,537 TRACE [f.p.e.t.t.TypeElasticsearchUpdater] createType([twitter/tweet])
[2017-03-12T15:59:32,333][INFO ][o.e.c.m.MetaDataMappingService] [Q3beqyO] [twitter/LT0TS5uFSpa4Is78h-8naA] create_mapping [tweet]
15:59:32,818 TRACE [f.p.e.t.t.TypeElasticsearchUpdater] /createType([twitter/tweet])
15:59:32,818 DEBUG [f.p.e.t.t.TypeElasticsearchUpdater] Type definition for [twitter]/[tweet] succesfully created.
15:59:32,818 TRACE [f.p.e.t.i.IndexSettingsReader] Reading [_settings.json] for [rss] in [es]...
15:59:32,818 TRACE [f.p.e.t.SettingsReader] Reading file [es/rss/_settings.json]...
15:59:32,819 TRACE [f.p.e.t.SettingsReader] Can not find [es/rss/_settings.json] in class loader.
15:59:36,115 DEBUG [f.p.e.t.i.IndexElasticsearchUpdater] Index [rss] doesn't exist. Creating it.
15:59:36,115 TRACE [f.p.e.t.i.IndexElasticsearchUpdater] createIndex([rss])
```

And suddenly 💥 

```
15:59:36,171 WARN  [o.s.c.s.ClassPathXmlApplicationContext] Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'esClient': Invocation of init method failed; nested exception is NoNodeAvailableException[None of the configured nodes were available: [{Q3beqyO}{Q3beqyOTRNmkkow3pXMK0g}{biMDuBQdQD2bxga9m2Cm6Q}{127.0.0.1}{127.0.0.1:9300}]]; nested: NodeDisconnectedException[[Q3beqyO][127.0.0.1:9300][indices:admin/create] disconnected];
15:59:36,173 INFO  [o.e.c.t.TransportClientNodesService] failed to get node info for {#transport#-1}{oBLQPybTT2akQRbmQF7o8g}{127.0.0.1}{127.0.0.1:9300}, disconnecting...
org.elasticsearch.transport.NodeDisconnectedException: [][127.0.0.1:9300][cluster:monitor/nodes/liveness] disconnected
```

If we look at elasticsearch logs:

```
[2017-03-12T15:59:24,349][INFO ][o.e.g.GatewayService     ] [Q3beqyO] recovered [0] indices into cluster_state
[2017-03-12T15:59:31,063][INFO ][o.e.c.m.MetaDataCreateIndexService] [Q3beqyO] [twitter] creating index, cause [api], templates [], shards [5]/[1], mappings []
[2017-03-12T15:59:32,333][INFO ][o.e.c.m.MetaDataMappingService] [Q3beqyO] [twitter/LT0TS5uFSpa4Is78h-8naA] create_mapping [tweet]
```

The process has been killed without any notice as we can see when elasticsearch-maven-plugin tries to stop it:

```
[INFO] --- elasticsearch-maven-plugin:5.7:stop (stop-elasticsearch) @ spring-elasticsearch ---
[INFO] Stopping Elasticsearch [com.github.alexcojocaru.mojo.elasticsearch.v2.InstanceConfiguration@6f2864c3[id=0,baseDir=/home/travis/build/dadoonet/spring-elasticsearch/target/elasticsearch0,httpPort=0,transportPort=0,pathData=<null>,pathLogs=<null>]]
[DEBUG] Using environment: null
[INFO] Elasticsearch[0]: Executing command '[kill, 1595]' in directory '/home/travis/build/dadoonet/spring-elasticsearch/target/elasticsearch0'
kill: No such process
[ERROR] Exception while stopping Elasticsearch
com.github.alexcojocaru.mojo.elasticsearch.v2.ElasticsearchSetupException: Elasticsearch [0]: Cannot execute command '[kill, 1595]' in directory '/home/travis/build/dadoonet/spring-elasticsearch/target/elasticsearch0'
	at com.github.alexcojocaru.mojo.elasticsearch.v2.util.ProcessUtil.executeScript(ProcessUtil.java:108)
	at com.github.alexcojocaru.mojo.elasticsearch.v2.util.ProcessUtil.executeScript(ProcessUtil.java:56)
	at com.github.alexcojocaru.mojo.elasticsearch.v2.StopMojo.execute(StopMojo.java:44)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:355)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:216)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:160)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
	at org.apache.commons.exec.DefaultExecutor.executeInternal(DefaultExecutor.java:404)
	at org.apache.commons.exec.DefaultExecutor.execute(DefaultExecutor.java:166)
	at com.github.alexcojocaru.mojo.elasticsearch.v2.util.ProcessUtil.executeScript(ProcessUtil.java:91)
	... 23 more
```


Let me know if you have any idea/workaround for this. Feel free to close if you don't think we can document anything regarding this problem.

--- a/README.md
+++ b/README.md
@@ -174,3 +174,14 @@ Each command has three parts, separated by colon.
 
 * To send a *DELETE* request to *http://localhost:9200/test\_index/test\_type/1* without content; note the colon at the end, for there is no JSON data in case of a DELETE.
 > DELETE:test\_index/test\_type/1:
+
+## FAQ
+
+**Node is killed when running in TravisCI**: When running your build job in [TravisCI](https://travis-ci.org/), it can happen that your node is being killed without any notice.
+To fix that you may have to modify the `.travis.yml` file as follows:
+
+```yml
+sudo: true
+before_script:
+  - sudo sysctl -w vm.max_map_count=262144
+```

